mypy_django_plugin/django/context.py

```diff
@@ -384,6 +384,10 @@ def resolve_lookup_into_field(
 
     def resolve_lookup_expected_type(self, ctx: MethodContext, model_cls: Type[Model], lookup: str) -> MypyType:
         query = Query(model_cls)
+        if lookup.startswith("pk") and query.get_meta().pk is None:
```

Comment:
`lookup.startswith("pk")` can match other column names, e.g. someone may abbreviate `private_key` as `pkey` which starts with `pk`

Does `solve_lookup_type()` fail in this situation? If not, the `lookup_parts` variable seems like better way to detect this.

Reply:
Yeah, true. While quite unlikely to also match on `.pk is None` we should still avoid it.

`solve_lookup_type()` is what's crashing. We could probably change the statement to:

```python
if lookup == "pk" or lookup.startswith("pk__") and query.get_meta().pk is None:
    ...
```

Reply:
While there's also the option to mimic Django

```python
lookup_splitted, _ = lookup.split(LOOKUP_SEP, 1)
if lookup_splitted == "pk" and query.get_meta().pk is None:
    ...
```

Reply:
Either of these look fine to me. I'll leave it up to you.

Reply:
I wen't with the first comment to avoid an import of`LOOKUP_SEP`